assets/app/scripts/filters/resources.js

```diff
@@ -89,6 +93,32 @@ angular.module('openshiftConsole')
       return tags.split(/\s*,\s*/);
     };
   })
+  .filter('imageAnnotationTags', function() {
+    return function(resource, /* optional */ tagName, /* optional */ annotationKey) {
+      if (!resource || !resource.spec || !resource.spec.tags) {
```

Comment:
can't you use `tagAnnotationFilter(resource, annotationKey || "tags", tagName)` to do the heavy lifting, then split here?